flycheck.el

```diff
@@ -2173,10 +2200,45 @@ into the verification results."
         (princ (make-string (- message-column (current-column)) ?\ ))
         (let ((message (flycheck-verification-result-message result))
               (face (flycheck-verification-result-face result)))
-          (insert (propertize message 'face face)))
+          ;; If face is nil, using propertize erases the face already contained
+          ;; by the message.  We don't want that, since this would remove the
+          ;; button face from the checker chain result.
+          (insert (if face (propertize message 'face face) message)))
         (princ "\n"))))
   (princ "\n"))
 
+(defun flycheck-get-next-checkers (checker)
+  "Return the immediate next checkers of CHECKER.
+
+This is a list of checker symbols.  The error levels of the
+`:next-checker' property are ignored."
+  (mapcar (lambda (c) (if (consp c) (cdr c) c))
+          (flycheck-checker-get checker 'next-checkers)))
+
+(defun flycheck-all-next-checkers (checker)
+  "Return all checkers that may follow CHECKER.
+
+Return the transitive closure of the next-checker relation.  The
+return value is a list of checkers, not including CHECKER."
+  (let ((next (flycheck-get-next-checkers checker)))
+    (seq-uniq (append next (seq-mapcat #'flycheck-all-next-checkers next)))))
```

Comment:
This is going to infloop if someone adds a checker as its own next-checker, right? I think our verification code will likely infloop as well, though, so this is not worse

Reply:
Yes, this assumes we don't have cycles.  Ideally we want to ensure there are no cycles by construction, but I'm not sure how to do that in `define-checker` and when the `:next-checkers` property is altered after.

Do you want me to harden this function to ensure we don't revisit checkers twice?

Reply:
@cpitclaudel I've changed the code to avoid visiting checkers twice.  Can you look?